src/backend/libpq/rangerrest.c

```diff
@@ -453,23 +453,30 @@ static int call_ranger_rest(CURL_HANDLE curl_handle, const char* request)
 		{
 			if (retry > 1)
 			{
-				elog(WARNING, "ranger plugin service from http://%s:%d/rps is unavailable : %s, try another http://%s:%d/rps\n",
+				/* Don't expose this warning message to client, just record in log.
+				 * The value of whereToSendOutput is DestRemote, so set it to DestNone
+				 * and set back after write a warning message in log file.
+				 */
+				CommandDest commandDest = whereToSendOutput;
+				whereToSendOutput = DestNone;
+				elog(WARNING, "ranger plugin service from http://%s:%d/rps is unavailable : %s, "
+						"trying ranger plugin service at http://%s:%d/rps\n",
```

Comment:
When master RPS doesn't work due to some reason, and hawq begins to talk with standby RPS, a warning message should be recorded in log file, so that administrators can solve the master RPS problem. elog(LOG, ...) won't expose to console(by default, client_min_sessages is NOTICE), but in this switch case, a warning should be recorded.